python/pyspark/sql/types.py

```diff
@@ -24,6 +24,7 @@
 import re
 import base64
 from array import array
+from functools import lru_cache
```

Comment:
Any ideas for Python 2.7 ?

Reply:
I think we should disable it in Python 2.x if we are going ahead with this.

Reply:
Or use backported library.
https://pypi.python.org/pypi/functools32

Reply:
I don't think the backport is not the builtin one. I'd not add this as a dependency for now.

Reply:
I added support for Python < 3.3.
What do you think ?